src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -510,7 +510,9 @@ private void CheckRequestBeforeSend(HttpRequestMessage request)
                 // Wait for the send request to complete, getting back the response.
                 response = async ?
                     await base.SendAsync(request, cts.Token).ConfigureAwait(false) :
+#pragma warning disable // Validate platform compatibility
                     base.Send(request, cts.Token);
+#pragma warning restore
```

Comment:
as `base.Send()` is called unconditionally this method should have `[UnsupportedOSPlatform("browser")]` attribute, but that is changing the signature of all `SendAsync()` overloads plus some `GetAsync()` overloads, suppressed for now

Reply:
> suppressed for now

Since all the places that call this with `async = false` are annotated with `[UnsupportedOSPlatform("browser")]`, I believe this should be "suppressed always". Right?

Reply:
> Since all the places that call this with async = false are annotated with [UnsupportedOSPlatform("browser")], I believe this should be "suppressed always". Right?

Yeah, though it's strange that all `async = false` are unsupported on browser but `async=true` ones are supported

Reply:
I assume that is because the underlying JavaScript APIs do not support synchronous HTTP. But someone on @lewing's or @steveisok's team should correct me if I'm wrong.

Reply:
@eerhardt That's correct. Since wasm is single threaded, the sync API's will never be able to yield back to the browser. It'll just hang if we enabled them.

Reply:
Would it be appropriate and acceptable to change this code to the following then?

``` csharp
if (async)
{
    await base.SendAsync(request, cts.Token).ConfigureAwait(false);
}
else
{
    Debug.Assert(!OperatingSystem.IsBrowser);
    base.Send(request, cts.Token);
}
```

If we're going to keep the `#pragma warning disable`, then we should specify the diagnostic id with `#pragma warning disable ca1416`.